javalib/src/main/scala/java/lang/StackTraceElement.scala

```diff
@@ -37,21 +113,18 @@ final class StackTraceElement(val getClassName: String,
 }
 
 private[lang] object StackTraceElement {
-  // symbol is "classname::methodName_T1_..._TN"
-  // where classname doesn't include colons and method name underscores.
-  def fromSymbol(symbol: String): StackTraceElement = {
-    val (className, methodName) =
-      symbol.indexOf("::") match {
-        case -1 =>
-          ("<none>", symbol)
-        case sep =>
-          symbol.indexOf("_", sep) match {
-            case -1 =>
-              (symbol.substring(0, sep), symbol.substring(sep + 2))
-            case end =>
-              (symbol.substring(0, sep), symbol.substring(sep + 2, end))
-          }
-      }
-    new StackTraceElement(className, methodName, null, 0)
+  private val cache =
+    collection.mutable.HashMap.empty[CUnsignedLong, StackTraceElement]
+  private def makeStackTraceElement(
+      cursor: Ptr[scala.Byte]): StackTraceElement = {
+    val name   = stackalloc[CChar](256)
+    val offset = stackalloc[scala.Byte](8)
+
+    unwind.get_proc_name(cursor, name, 256, offset)
+    new StackTraceElement(name)
   }
+  def cached(cursor: Ptr[scala.Byte],
```

Comment:
Lets also add a comment that explains why and how are stack trace elements cached.